src/main/scala/org/eiennohito/grpc/stream/impl/client/GrpcClientHandler.scala

```diff
@@ -92,8 +96,12 @@ class GrpcClientHandler[Req, Resp](
       private def handleClose(t: Status, m: Metadata) = {
         trls.success(m)
         if (t.isOk) {
-          cmpl.success(Done)
-          completeStage()
+          if (buffer.isEmpty) {
+            cmpl.success(Done)
```

Comment:
Should `cmpl.success(Done)` be called immediately and only defer the `completeStage()` call?